### ISSUE TITLE
Unable to configure with Redis Socket

### ISSUE THREAD

**Issue by 98h398hrpohpoai** [firefly-iii/firefly-iii#3642]
**Bug description**
I am running Firefly III version 5.2.8, and my problem is:

Unable to use redis when running it as a socket rather than tcp address. 
Produces the following error when trying to do so: Failed to parse address "/run/redis/redis.sock:" [tcp:///run/redis/redis.sock:] 

**Steps to reproduce**
Configure redis as socket rather than tcp address like so:

CACHE_DRIVER=redis
SESSION_DRIVER=redis
REDIS_HOST=/run/redis/redis-server.sock
REDIS_PASSWORD=null
REDIS_PORT=
REDIS_DB="0"
REDIS_CACHE_DB="1"

**Extra info**
Brief summary on using redis sockets: https://guides.wp-bullet.com/how-to-configure-redis-to-use-unix-socket-speed-boost/
In short they can provide up to a 25% performance boost in addition to being more secure.
Looks like the error results because the redis host is automatically parsed as a tcp protocol instead of allowing an option for unix, but that's just speculation.

**Comment by JC5** [comment]
I'll pick it up as an enhancement :+1:
